xdsl/tools/xdsl_tool.py

```diff
@@ -0,0 +1,218 @@
+import argparse
+import os
+import sys
+from typing import IO, Callable
+
+from xdsl.backend.riscv.lowering.riscv_arith_lowering import RISCVLowerArith
+from xdsl.dialects.affine import Affine
+from xdsl.dialects.arith import Arith
+from xdsl.dialects.builtin import Builtin, ModuleOp
+from xdsl.dialects.cf import Cf
+from xdsl.dialects.cmath import CMath
+from xdsl.dialects.experimental.dmp import DMP
+from xdsl.dialects.experimental.fir import FIR
+from xdsl.dialects.experimental.math import Math
+from xdsl.dialects.func import Func
+from xdsl.dialects.gpu import GPU
+from xdsl.dialects.irdl.irdl import IRDL
+from xdsl.dialects.linalg import Linalg
+from xdsl.dialects.llvm import LLVM
+from xdsl.dialects.memref import MemRef
+from xdsl.dialects.mpi import MPI
+from xdsl.dialects.pdl import PDL
+from xdsl.dialects.printf import Printf
+from xdsl.dialects.riscv import RISCV
+from xdsl.dialects.riscv_func import RISCV_Func
+from xdsl.dialects.scf import Scf
+from xdsl.dialects.snitch import Snitch
+from xdsl.dialects.snitch_runtime import SnitchRuntime
+from xdsl.dialects.stencil import Stencil
+from xdsl.dialects.test import Test
+from xdsl.dialects.vector import Vector
+from xdsl.frontend.passes.desymref import DesymrefyPass
+from xdsl.frontend.symref import Symref
+from xdsl.ir import Dialect, MLContext
+from xdsl.parser import Parser
+from xdsl.passes import ModulePass
+from xdsl.transforms.dead_code_elimination import DeadCodeElimination
+from xdsl.transforms.experimental.convert_stencil_to_ll_mlir import (
+    ConvertStencilToLLMLIRPass,
+)
+from xdsl.transforms.experimental.dmp.stencil_global_to_local import (
+    GlobalStencilToLocalStencil2DHorizontal,
+    LowerHaloToMPI,
+)
+from xdsl.transforms.experimental.stencil_shape_inference import (
+    StencilShapeInferencePass,
+)
+from xdsl.transforms.experimental.stencil_storage_materialization import (
+    StencilStorageMaterializationPass,
+)
+from xdsl.transforms.lower_mpi import LowerMPIPass
+from xdsl.transforms.lower_riscv_func import LowerRISCVFunc
+from xdsl.transforms.lower_snitch import LowerSnitchPass
+from xdsl.transforms.lower_snitch_runtime import LowerSnitchRuntimePass
+from xdsl.transforms.mlir_opt import MLIROptPass
+from xdsl.transforms.printf_to_llvm import PrintfToLLVM
+from xdsl.transforms.reconcile_unrealized_casts import ReconcileUnrealizedCastsPass
+from xdsl.transforms.riscv_register_allocation import RISCVRegisterAllocation
```

Comment:
Same here btw.

Reply:
it seems I have forgotten to add a comment :thinking: 

This is concerning the on lines 7-31 (which I will readd in a moment...)

Reply:
Will rename those, I'm assuming you and @math-fehr came to an agreement to keep the dialect ones as is for now! Tell me if I misinterpreted :slightly_smiling_face: